ec/ecresource/deploymentresource/read.go

```diff
@@ -174,6 +178,31 @@ func (r *Resource) read(ctx context.Context, id string, state *deploymentv2.Depl
 
 	deployment.NullifyUnusedEsTopologies(ctx, baseElasticsearch)
 
+	if !deployment.HasNodeTypes() {
+		// The MigrateDeploymentTemplate request can only be performed for deployments that use node roles.
+		// We'll skip this logic for deployments with node types.
+		migrateTemplateRequest, err := r.client.V1API.Deployments.MigrateDeploymentTemplate(
+			deployments.NewMigrateDeploymentTemplateParams().WithDeploymentID(deployment.Id).WithTemplateID(deployment.DeploymentTemplateId),
+			r.client.AuthWriter,
+		)
+
+		// Store migrate request in private state
+		if readResponse != nil {
```

Comment:
Shall we do this after the error check in lines 194-197?

Reply:
Or the idea is that it should cache `nil` in case of an error?

Reply:
Addressed in https://github.com/elastic/terraform-provider-ec/pull/755/commits/59a1c9a4e9bdebd7e6bd78656cb465b39550abb1.